src/main/java/LinkedList/MyLinkedList.java

```diff
@@ -0,0 +1,156 @@
+package LinkedList;
+
+public class MyLinkedList<T> {
+    private Node<T> head;
+    private int size;
+
+    MyLinkedList() {
+        head = null;
+        size = 0;
+    }
+
+    public void addHead(T value) {
+        Node<T> newNode = new Node<T>(value);
+        newNode.next = head;
+        head = newNode;
+        size++;
+    }
+
+    public void addTail(T value) {
+        Node<T> newNode = new Node<T>(value);
+        if (head == null) {
+            head = newNode;
+        } else {
+            Node<T> temp = head;
+            while (temp.next != null) {
+                temp = temp.next;
+            }
+            temp.next = newNode;
+        }
+        size++;
+    }
+
+    public T removeHead() {
+        if (head == null) {
+            return null;
+        }
+        T data = head.data;
+        head = head.next;
+        size--;
+        return data; //silinen deyeri return edir
+    }
+
+    public T removeTail() {
+        if (head == null) {
+            return null;
+        }
+        if (head.next == null) {
+            T data = head.data;
+            head = null;
+            size--;
+            return data;
+        }
+        Node<T> temp = head;
+        while (temp.next.next != null) {
+            temp = temp.next;
+        }
+        T data = temp.next.data;
+        temp.next = null;
+        size--;
+        return data;
+    }
+
+    public void insert(int index, T value) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index is out of bounds");
+        }
+        Node<T> newNode = new Node(value);
+        if (index == 0) {
+            addHead(value);
+        } else {
+            Node<T> temp = head;
+            for (int i = 0; i < index - 1; i++) {
+                temp = temp.next;
+            }
+            newNode.next = temp.next;
+            temp.next = newNode;
+        }
+        size++;
```

Comment:
if index == 0 it will call method addHead() and inside this method we already increment size